packages-exp/auth-exp/src/core/auth/auth_impl.ts

```diff
@@ -68,25 +75,73 @@ export class AuthImpl implements Auth {
   ) {}
 
   _initializeWithPersistence(
-    persistenceHierarchy: Persistence[]
+    persistenceHierarchy: Persistence[],
+    popupRedirectResolver?: externs.PopupRedirectResolver
   ): Promise<void> {
     return this.queue(async () => {
       this.persistenceManager = await PersistenceUserManager.create(
         this,
         persistenceHierarchy
       );
 
-      const storedUser = await this.persistenceManager.getCurrentUser();
-      // TODO: Check redirect user, if not redirect user, call refresh on stored user
-      if (storedUser) {
-        await this.directlySetCurrentUser(storedUser);
-      }
+      await this.initializeCurrentUser(popupRedirectResolver);
 
       this._isInitialized = true;
       this.notifyAuthListeners();
     });
   }
 
+  private async initializeCurrentUser(
+    popupRedirectResolver?: externs.PopupRedirectResolver
+  ): Promise<void> {
+    const storedUser = await this.assertedPersistence.getCurrentUser();
+    if (!storedUser) {
+      return this.directlySetCurrentUser(storedUser);
+    }
+
+    if (!storedUser._redirectEventId) {
+      // This isn't a redirect user, we can reload and bail
+      return this.reloadAndSetCurrentUserOrClear(storedUser);
+    }
+
+    assert(popupRedirectResolver, this.name, AuthErrorCode.ARGUMENT_ERROR);
```

Comment:
we may want a more user friendly error here

Reply:
on the one hand: yes. on the other: this promise is not returned to the user so they'll never see it :grimacing: